### PR TITLE
Mongo Middleware

### LINKED ISSUE #3: Middleware: mongodb
Create mongodb middleware.

Reference original source, though improvements may be warranted.

### PR DESCRIPTION
**Pull request by therealklanni**
Closes #3

### CODE REVIEW

File: src/server/middleware/mongo/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+db.on('error', console.error.bind(console, 'connection error: '))
+
+db.once('open', () => {
+  debug('MongoDB connected')
+})
+
+export default () => function * mongodb (next) {
+  this.db = db
+  this.db.User = User;
```
Comment (troym9731) [inline src/server/middleware/mongo/index.js]: What is this ';' character?

File: src/server/middleware/request-logger/index.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import logger from '../../../common/lib/logger'
 const debug = logger('SIR:request')
+const debugBody = logger('SIR:request:body')
 
 import chalk from 'chalk'
 import perf from '../../lib/perf'
 
-export default function * requestLogger (next) {
+export default () => function * requestLogger (next) {
```
Comment (troym9731) [inline src/server/middleware/request-logger/index.js:8]: function *requestLogger(next), or ESLint will complain later on

File: src/server/middleware/response-timer/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import perf from '../../lib/perf'
 
-export default function *responseTime (next) {
+export default () => function *responseTime (next) {
```
Comment (troym9731) [inline src/server/middleware/response-timer/index.js:3]: Also *responseTime(next)

File: src/server/middleware/mongo/index.js (hunk 1 above)
Comment (therealklanni) [inline src/server/middleware/mongo/index.js]: haha. I have to use them at work, so sometimes my fingers get confused ;p

File: src/server/middleware/mongo/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const db = mongoose.connection
+
+db.on('error', console.error.bind(console, 'connection error: '))
+
+db.once('open', () => {
+  debug('MongoDB connected')
+})
+
+export default () => function * mongodb (next) {
```
Comment (troym9731) [inline src/server/middleware/mongo/index.js:17]: Found another instance *mongodb(next)

**Pull request merged**